multisig-ism-metadata/src/main.sw

```diff
@@ -0,0 +1,72 @@
+library metadata;
+
+use std::{b512::B512, bytes::Bytes, hash::keccak256, vm::evm::evm_address::EvmAddress};
+
+use bytes_extended::*;
+
+pub struct MultisigMetadata {
+    root: b256,
+    index: u32,
+    mailbox: b256,
+    proof: [b256; 32],
+    threshold: u8,
+    signatures: Vec<B512>,
+    validators: Vec<EvmAddress>,
+}
+
+const U8_BYTE_COUNT = 1u64;
+
+pub fn domain_hash(origin: u32, mailbox: b256) -> b256 {
+    let suffix = "HYPERLANE";
+    let suffix_len = 9;
+
+    let mut bytes = Bytes::with_length(U32_BYTE_COUNT + B256_BYTE_COUNT + suffix_len);
```

Comment:
any take on pulling out the sum into a const? Here and checkpoint_hash
Don't feel too strongly here but I know that at least in Solidity I'd prefer as little unnecessary arithmetic as possible, so I'm kind of bringing that here

Reply:
kinda assume the compiler does this automagically